fendermint/app/settings/src/lib.rs

```diff
@@ -93,6 +97,52 @@ pub struct BroadcastSettings {
     pub max_retries: u8,
 }
 
+#[derive(Debug, Deserialize, Clone)]
+pub struct TopDownConfig {
+    /// The number of blocks to delay before reporting a height as final on the parent chain.
+    /// To propose a certain number of epochs delayed from the latest height, we see to be
+    /// conservative and avoid other from rejecting the proposal because they don't see the
+    /// height as final yet.
+    pub chain_head_delay: BlockHeight,
+    /// Parent syncing cron period, in seconds
+    pub polling_interval_secs: u64,
+    /// Top down exponential back off retry base
+    pub exponential_back_off_secs: u64,
+    /// The max number of retries for exponential backoff before giving up
+    pub exponential_retry_limit: usize,
+    /// The ipc provider url
+    pub ipc_provider_url: String,
+    /// The parent registry address
+    #[serde(deserialize_with = "deserialize_eth_address_from_str")]
+    pub registry_address: Address,
```

Comment:
The gateway and the registry do not have stable addresses on Lotus, right?

Reply:
Correct, that is why we need to offer a config for that (for L2+ subnets this will be easier as we can point to the deterministic address by Fendermint)